PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/FunctionCoreToolsCombobox.java

```diff
@@ -100,26 +102,28 @@ private void openAzureSettingsPanel() {
         openSettingsAction.getHandler(null, event).accept(null, event); // Open Azure Settings Panel sync
     }
 
-    public void reset() {
-        funcCoreToolsPathList.clear();
-        this.clear();
-        funcCoreToolsPathList.addAll(loadHistory());
-        funcCoreToolsPathList.addAll(FunctionCliResolver.resolve());
-
-        final String valueFromAzConfig = Azure.az().config().getFunctionCoreToolsPath();
-        if (StringUtils.isNotBlank(valueFromAzConfig) && Files.exists(Path.of(valueFromAzConfig))) {
-            funcCoreToolsPathList.add(valueFromAzConfig);
-        }
-        funcCoreToolsPathList.forEach(this::addItem);
+    private String getDefaultFuncPath() {
+        final String functionCoreToolsPath = Azure.az().config().getFunctionCoreToolsPath();
+        return StringUtils.isNotEmpty(functionCoreToolsPath) && Files.exists(Path.of(functionCoreToolsPath)) ? functionCoreToolsPath : null;
+    }
 
-        if (includeSettings) {
-            this.addItem(OPEN_AZURE_SETTINGS);
-        }
-        if (StringUtils.isNotBlank(valueFromAzConfig)) {
-            this.setValue(valueFromAzConfig);
-        } else {
-            this.setSelectedIndex(-1);
+    @Nonnull
+    @Override
+    protected List<? extends String> loadItems() throws Exception {
+        final List<String> result = new ArrayList<>();
+        result.addAll(loadHistory());
+        result.addAll(FunctionCliResolver.resolve());
+        result.add(getDefaultFuncPath());
+        result.add(includeSettings ? OPEN_AZURE_SETTINGS : null);
+        return result.stream().filter(Objects::nonNull).distinct().sorted().collect(Collectors.toList());
```

Comment:
it would be better to make `defaultFuncPath` take the first position.

Reply:
will call `reset` during constructor, which will select `defaultFuncPath` by default